android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -568,6 +573,22 @@ protected void onSafeCreate(@Nullable Bundle savedInstanceState)
      */
     if (Map.isEngineCreated())
       onRenderingInitializationFinished();
+
+    // Force native system locale initialization at app start-up.
+    Language.nativeRefreshSystemLocale();
+
+    // Setup BroadcastReceiver to receive changes of system locale.
+    mLocaleChangeReceiver = new BroadcastReceiver()
```

Comment:
> This receiver probably will be more appropriate in MwmApplication instead of MwmActivity.

@rtsisyk: Moved Locale initialization and `BroadcastReceiver` to `MwmApplication`.